scripts/start

```diff
@@ -0,0 +1,162 @@
+    alias: 'H',
+    type: 'string',
+    default: process.env.HOST || process.env.HOSTNAME || '0.0.0.0',
+    describe: 'The host to bind to',
+})
+.option('port', {
+    alias: 'p',
+    type: 'number',
+    default: Number(process.env.PORT) || 3000,
+    describe: 'The port to bind to',
+})
+.option('gzip', {
+    alias: 'gz',
+    type: 'boolean',
+    default: process.env.GZIP !== '0',
+    describe: 'Enable or disable gzip compression',
+})
+.option('reporter', {
+    type: 'string',
+    describe: 'Any of the planify\'s reporters',
+})
+.example('$0', 'Serves the last built application')
+.example('$0 --port 8081', 'Serves the last built application on port 8081')
+.argv;
+
+// ---------------------------------------------------------
+// Functions
+// ---------------------------------------------------------
+
+function prepare(data) {
+    // Force ENV to production
+    process.env.NODE_ENV = 'production';
+    process.env.BABEL_ENV = 'production';
+
+    // Read the manifest & import the server code
+    data.buildManifest = readBuildManifest();
+
+    const serverFile = `${publicDir}/build/${data.buildManifest.server.file}`;
+
+    try {
+        data.exports = require(serverFile);
+    } catch (err) {
+        err.detail = `This error happened when loading the server file at ${path.relative('', serverFile)}`;
+        err.hideStack = false;
+        throw err;
+    }
+
+    process.stdout.write(`Build hash: ${data.buildManifest.hash}\n`);
+    process.stdout.write(`Server build hash: ${data.buildManifest.server.hash}\n`);
+}
+
+async function runServer(data) {
+    const { host, port, gzip } = argv;
+    const app = express();
+
+    // Configure express app
+    app.set('etag', false);  // Not necessary by default
+    app.set('x-powered-by', false);  // Remove x-powered-by header
+
+    // Enable gzip compression
+    gzip && app.use('/', compression());
+
+    // Public files in /build have the following pattern: <file>.<hash>.<extension>
+    // Therefore it's safe to cache them indefinitely
+    app.use('/build', express.static(`${publicDir}/build`, {
+        maxAge: 31557600000, // 1 year
+        immutable: true,     // No conditional requests
+        etag: false,         // Not necessary
+        index: false,        // Disable directory listing
+        fallthrough: false,  // Ensure that requests to /build do not propagate to other middleware
+    }));
+
+    // The rest of the public files are served using a more modest approach using etags
+    app.use(express.static(publicDir, {
+        index: false,
+    }));
+
+    // There's no compilation to be done here, just copy stuff to `res.locals`
+    app.use((req, res, next) => {
+        res.locals.exports = data.exports;
+        res.locals.buildManifest = data.buildManifest;
+        next();
+    });
+
+    // If it's not a public file, render the app!
+    app.use(
+        // There's no compilation to be done, just set the `res.locals`
+        (req, res, next) => {
```

Comment:
Isn't this block duplicated? Locals are already being copied above

Reply:
Yep it is. removed!